test/flutter_tools/artifacts.dart

```diff
@@ -0,0 +1,387 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```

Comment:
A drive by comment here as I have not kept up with this PR, but this is bringing in *a lot* of flutter_tools code, including code that I wouldn't think would be related to driving a `flutter-tester` device based test. What are some other options you've looked at, or, what's driving including so much code?

Reply:
I suppose we could look into pruning test_driver.dart down to include only what we are using. 
That may eliminate some of the dependencies, and if we need to add stuff back in in the future, we could do so as needed.

Reply:
I agree that pruning test_driver.dart down a bit is probably the best option.

Another option would be to use a path dependency to import flutter_tools even though it is not published as a package on pub. The risk is that flutter_tools could easily have breaking changes that would only impact us.

Reply:
OK, sounds like this is bringing in `flutter_tools/test_driver.dart` and related files. The primary thing we need from the flutter sdk is the actual `flutter-tester` test device. We can then write a wrapper class - similar to `test_driver.dart` - which starts up a `flutter run --machine -d flutter-tester` process and speaks JSON RPC to it over stdout.